pkg/controllers/binding/binding_controller_test.go

```diff
@@ -0,0 +1,592 @@
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: *spec,
+	}
+}
+
+func newResourceBindingSpec(resource *workv1alpha2.ObjectReference, replicas int32, clusters []workv1alpha2.TargetCluster) *workv1alpha2.ResourceBindingSpec {
+	return &workv1alpha2.ResourceBindingSpec{
+		Resource: *resource,
+		Replicas: replicas,
+		Clusters: clusters,
+	}
+}
+
+func newObjectReference(apiVersion string, kind string, namespace string, name string) *workv1alpha2.ObjectReference {
+	return &workv1alpha2.ObjectReference{
+		APIVersion: apiVersion,
+		Kind:       kind,
+		Namespace:  namespace,
+		Name:       name,
+	}
+}
+
+func newWork(bindingNamespace string, bindingName string, clusterName string, workName string) *workv1alpha1.Work {
+	esname, err := names.GenerateExecutionSpaceName(clusterName)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	referenceKey := names.GenerateBindingReferenceKey(bindingNamespace, bindingName)
+	ls := labels.Set{workv1alpha2.ResourceBindingReferenceKey: referenceKey}
+
+	annotations := make(map[string]string)
+	annotations[workv1alpha2.ResourceBindingNameAnnotationKey] = bindingName
+	annotations[workv1alpha2.ResourceBindingNamespaceAnnotationKey] = bindingNamespace
+
+	return &workv1alpha1.Work{
+		TypeMeta:   metav1.TypeMeta{APIVersion: workv1alpha1.SchemeGroupVersion.String(), Kind: workv1alpha1.ResourceKindWork},
+		ObjectMeta: metav1.ObjectMeta{Name: workName, Namespace: esname, Labels: ls, Annotations: annotations},
+	}
+}
+
+// Compare the real work and expected work
+func (f *fixture) isWorkEqual() (res bool) {
+	workList := &workv1alpha1.WorkList{}
+
+	err := f.client.List(context.TODO(), workList)
+	if err != nil {
+		f.t.Fatal(err)
+	}
+
+	works := workList.Items
+
+	defer func() {
+		if !res {
+			f.t.Errorf("expectedWorks: %v, but got: %v", f.expectedWorks, works)
+		}
+	}()
+
+	if len(f.expectedWorks) != len(works) {
+		f.t.Errorf("len(expectedWorks) == %v, but got len(actualWorks) == %v\n", len(f.expectedWorks), len(works))
+		return false
+	}
+
+	flag := false
+
+	for i := range works {
+		work := works[i]
+		for j := range f.expectedWorks {
+			expectedWork := f.expectedWorks[j]
+			if work.Namespace == expectedWork.Namespace && work.Name == expectedWork.Name {
+				flag = true
+				break
+			}
+		}
+		if !flag {
+			f.t.Errorf("Unexpected Work: %v", work)
+			return false
+		}
+		flag = false
+	}
+
+	return true
+}
+
+func TestSyncResourceBindingCreatesWork(t *testing.T) {
+	f := newFixture(t)
+	controller := f.newController()
+
+	d := deploymentObject
+	rb := resourceBindingObject
+	c := clustersObject
+	w := worksObject
+
+	f.deployments = append(f.deployments, &d)
+	f.resourceBindings = append(f.resourceBindings, &rb)
+	f.clusters = append(f.clusters, &c[0], &c[1], &c[2])
+
+	f.expectedWorks = append(f.expectedWorks, w...)
+
+	f.objectResourceInit()
+
+	// Wait for informerManager's cache sync
+	_ = f.informerManager.Lister(appsv1.SchemeGroupVersion.WithResource("deployments"))
+	f.informerManager.Start()
```

Comment:
Does this step is needed?

The fakeClient may not be suitable for the informer: https://github.com/kubernetes/client-go/blob/c81636cd4415f0858a99c2cc18023740df7aa15f/examples/fake-client/main_test.go#L78-L81

Reply:
This step is needed for informerManager to sync deployments object.The informanager will not have the "deployments" informer and object store without this step. 
According to https://github.com/kubernetes/client-go/blob/c81636cd4415f0858a99c2cc18023740df7aa15f/examples/fake-client/main_test.go#L74-L77,it doesn't support resource version,so we can add objects to fake client before the informer created.Then the informer will use List method to get all objects.